docs/blogs/conformance.html

```diff
@@ -0,0 +1,240 @@
+        log.extend(dialer_ack.encode());
+
+        let (mut listener_tx, mut listener_rx) = listen_end(listener_state, dialer_ack).unwrap();
+
+        // Generate a random message to send to the listener from the dialer.
+        let mut random_msg = vec![0u8; rng.gen_range(0..256)];
+        rng.fill(&mut random_msg[..]);
+        log.extend(random_msg.encode());
+
+        let dialer_ciphertext = dialer_tx.send(random_msg.as_slice()).unwrap();
+        assert_ne!(dialer_ciphertext, random_msg);
+        log.extend(dialer_ciphertext.encode());
+
+        let received_msg = listener_rx.recv(&dialer_ciphertext).unwrap();
+        assert_eq!(received_msg, random_msg);
+        log.extend(received_msg.encode());
+
+        // -- snip --
+
+        log
+    }
+}
+
+conformance_tests! {
+    Handshake =&gt; 4096,
+}
+            </code>
+        </pre>
+
+        <h2>Commonware's Commitment to Stability</h2>
+        <p>Commonware is coming up on its first stable release, and we are committed to providing stable and reliable software for our users. By adopting conformance testing as a standard practice, we can ensure that our libraries' functionality remains robust and trustworthy, even as we continue to innovate on and evolve the underlying mechanisms.</p>
+        <p>In our codebase today, <code>conformance</code> covers:</p>
+        <ul>
+            <li>Wire formats (<code>codec</code>)</li>
+            <li>Database operation semantics (on-disk contents, read/write order, etc.) (<code>storage</code>)</li>
+            <li>Cryptographic mechanism semantics (operation order, input/output consistency, etc.) (<code>cryptography</code>)</li>
+        </ul>
+        <p>With more primitives to follow suit. Each time a breaking change to the conformance specification is introduced, our CI fails, and we are forced to explicitly acknowledge the change by updating the conformance commitments and adding a "<a href="https://github.com/commonwarexyz/monorepo/pulls?q=is%3Aopen+is%3Apr+label%3Abreaking-format">breaking-format</a>" or "<a href="https://github.com/commonwarexyz/monorepo/pulls?q=is%3Aopen+is%3Apr+label%3Abreaking-api">breaking-api</a>" label to the change. This process not only helps us catch bugs early but also serves as a trail of breadcrumbs for us to use while authoring releases that include intentional breakages.</p>
+        <p>Alongside our other testing strategies (<a href="https://commonware.xyz/blogs/commonware-runtime"><code>runtime</code> for deterministic simulations</a>, <a href="https://github.com/commonwarexyz/monorepo/blob/HEAD/FUZZING.md">prolonged fuzzing campaigns</a>, <a href="https://alto.commonware.xyz">persistent devnets</a>, and <a href="https://github.com/commonwarexyz/monorepo/actions/workflows/slow.yml">an extensive suite of unit + integration tests</a>), conformance testing is a powerful tool in our arsenal to ensure that we deliver high-quality software that meets the stability needs of our users.</p>
```

Comment:
> "that we deliver high-quality software that meets the stability needs of our users."

I think we can find a way to make this sound less corpo (software, no matter how neat, is only useful if reliable)